### PR TITLE
fix several newline problems

### PR DESCRIPTION
**Pull request by narimiran**

### CODE REVIEW

**Comment by ghost** [comment]
Is there any reason checks are with `doAssert` and not `assert`?

File: lib/pure/asyncftpclient.nim [hunk 1]
```diff
@@ -146,7 +146,11 @@ proc send*(ftp: AsyncFtpClient, m: string): Future[TaintedString] {.async.} =
   ## Send a message to the server, and wait for a primary reply.
   ## ``\c\L`` is added for you.
   ##
+  ## You need to make sure that the message ``m`` doesn't contain any newline
+  ## characters. Failing to do so will raise ``AssertionDefect``.
```
Comment (Varriount) [inline lib/pure/asyncftpclient.nim:150]: In non-release mode, right? Aren't assertions turned off for release mode?

Comment (ghost) [reply lib/pure/asyncftpclient.nim:150]: @Varriount doAssert assertions are always enabled

File: lib/pure/httpclient.nim [hunk 2]
```diff
@@ -245,7 +245,8 @@ proc contentLength*(response: Response | AsyncResponse): int =
   ##
   ## A ``ValueError`` exception will be raised if the value is not an integer.
   var contentLengthHeader = response.headers.getOrDefault("Content-Length")
-  return contentLengthHeader.parseInt()
+  result = contentLengthHeader.parseInt()
```
Comment (Araq) [inline lib/pure/httpclient.nim:248]: Use ``parseSaturatedNatural`` because that cannot raise an overflow exception.

Comment (narimiran) [reply lib/pure/httpclient.nim:248]: Yes, but that changes the documented behaviour:

> A ``ValueError`` exception will be raised if the value is not an integer.

```nim
import parseutils, strutils

var a = "123.45"

var b: int
discard parseSaturatedNatural(a, b)
echo b # ==> 123

var c = parseInt(a)
echo c # ==> ValueError
```

Comment (Araq) [reply lib/pure/httpclient.nim:248]: Ok...

**Pull request merged**